app/javascript/packages/form-steps/form-steps.tsx

```diff
@@ -256,27 +267,23 @@ function FormSteps({
   }
 
   const unknownFieldErrors = activeErrors.filter((error) => !fields.current[error.field]?.element);
-  const hasUnresolvedFieldErrors =
-    activeErrors.length && activeErrors.length > unknownFieldErrors.length;
 
   /**
    * Increments state to the next step, or calls onComplete callback if the current step is the last
    * step.
    */
-  const toNextStep: FormEventHandler = (event) => {
-    event.preventDefault();
-
-    // Don't proceed if field errors have yet to be resolved.
-    if (hasUnresolvedFieldErrors) {
+  function toNextStep() {
+    // Don't proceed if initial active errors have yet to be resolved.
+    if (activeErrors.length && activeErrors === initialActiveErrors) {
```

Comment:
Kinda an edge case, but I spotted an issue with this revised logic. When a user uploads invalid documents where we need them to upload both the front and back, we expect that they would not be able to submit until they replace _both_ front and back. With the revised implementation here, they would have been allowed to submit after replacing any of the values.

I added a failing regression test case in d02093076 and am working to revise this a bit, hopefully returning more to the previous logic.

Reply:
Unsure if it's the same bug... but when I was testing doc auth in staging yesterday, I accidentally set the front image for both front and back, so the real fix was to only replace back image, but I had to replace both images in order for the form to let me go

Reply:
> Unsure if it's the same bug... but when I was testing doc auth in staging yesterday, I accidentally set the front image for both front and back, so the real fix was to only replace back image, but I had to replace both images in order for the form to let me go

I think in the majority of cases we're expecting the user will need to replace both.

Reply:
> Unsure if it's the same bug... but when I was testing doc auth in staging yesterday, I accidentally set the front image for both front and back, so the real fix was to only replace back image, but I had to replace both images in order for the form to let me go

Also, were you using real images which would normally pass, since staging would be running the actual vendor code?

Reply:
yup staging, real vendors, real images that would have passed if I put the right back image in the right field

Reply:
I checked the logs and it looks like the back-end had reported an error on both sides of the image in that attempt.

```
properties.event_properties.errors.back.0 | Please add a new image
properties.event_properties.errors.front.0 | Please add a new image
properties.event_properties.errors.general.0 | We couldn’t verify your ID. Try taking new pictures.
```